lumicks/pylake/kymotracker/tests/test_derived_quantities/test_msd.py

```diff
@@ -318,3 +319,137 @@ def test_weighted_variance_error_case():
 
     with pytest.raises(ValueError, match="Mean and count arrays must be the same size"):
         weighted_mean_and_sd(np.array([2, 3]), np.array([2, 3, 4]))
+
+
+@pytest.mark.parametrize(
+    "frame_idx,position,max_lag,ref_lags,ref_rhos,ref_samples",
+    [
+        [[1, 2, 3, 4], [1, 2, 3, 4], 1000, [1, 2, 3], [1.0, 4.0, 9.0], [3, 2, 1]],
+        [[1, 2, 4], [1, 2, 4], 1000, [1, 2, 3], [1.0, 4.0, 9.0], [1, 1, 1]],
+        [[1, 2, 5], [1, 2, 4], 1000, [1, 3, 4], [1.0, 4.0, 9.0], [1, 1, 1]],
+        [[1, 2, 5, 6], [1, 2, 4, 5], 1000, [1, 3, 4, 5], [1.0, 4.0, 9.0, 16.0], [2, 1, 2, 1]],
+        [
+            [1, 2, 5, 6],
+            [1.5, 0.5, 3.0, 5.5],
+            1000,
+            [1, 3, 4, 5],
+            [3.625, 6.25, 13.625, 16.0],
+            [2, 1, 2, 1],
+        ],
+    ],
+)
```

Comment:
This is an interresting one actually. I think `max_lag` currently puts a limit on the maximum _number_ of lags included, rather than the maximum lag. This is probably a cornercase, and one that doesn't happen frequently in real scenarios, but it still poses a question. What to do when the two disagree? The results from Michalet are likely not valid in this case anyhow and the resulting estimates are probably less than accurate.

That said, I think `max_lag` should probably be the maximum lag included, and not the number of lags. Therefore I will make a separate commit at the very end that fixes this bug.

Reply:
Actually, looking at it a bit further, for consistency sake, I think it would be better to keep `max_lag` as is. Typically they refer to the number of lags included and in the docstrings on `KymoTrackGroup` itself it is clearly described as such. I have added a test for this and added a comment.